src/core/raster/qgsrasterlayerelevationproperties.cpp

```diff
@@ -150,15 +222,25 @@ void QgsRasterLayerElevationProperties::setBandNumber( int band )
   emit profileGenerationPropertyChanged();
 }
 
-double QgsRasterLayerElevationProperties::elevationForPixelValue( int band, double pixelValue ) const
+QgsDoubleRange QgsRasterLayerElevationProperties::elevationRangeForPixelValue( int band, double pixelValue ) const
 {
-  if ( !mEnabled )
-    return std::numeric_limits< double >::quiet_NaN();
+  if ( !mEnabled || std::isnan( pixelValue ) )
+    return QgsDoubleRange();
 
-  if ( band != mBandNumber )
-    return std::numeric_limits< double >::quiet_NaN();
+  switch ( mMode )
+  {
+    case Qgis::RasterElevationMode::FixedElevationRange:
+      return mFixedRange;
 
-  return pixelValue * mZScale + mZOffset;
+    case Qgis::RasterElevationMode::RepresentsElevationSurface:
+    {
+      if ( band != mBandNumber )
+        return QgsDoubleRange();
+
+      return QgsDoubleRange( pixelValue * mZScale + mZOffset, pixelValue * mZScale + mZOffset );
```

Comment:
Micro optimization: avoid counting it twice by using a const double?

Reply:
Ok, done